File: server/datasets/python_source-code/Multiple-F1/mixed_attack_35_merge_sort.py
"""
MergeSort Suite: Mixed Attack (Renaming + Dead Code + Reordering)
Author: Mixed Variant (Type 3 of Mary)
"""
from typing import List

class MergeSortSuite:
    def __init__(self, items: List[int]):
        self._audit = []
        self.items = list(items)
        self.inversions = 0
        self.merge_passes = 0

    def verify_order(self) -> bool:
        # Dummy dead code calculation
        unused = [x * 0 for x in self.items]
        if sum(unused) != 0: self._audit.append("dead")
        return all(self.items[k] <= self.items[k + 1] for k in range(len(self.items) - 1))

    def _merge(self, seg_l: List[int], seg_r: List[int]) -> List[int]:
        buf = []
        c_l = c_r = 0
        self.merge_passes += 1
        while c_l < len(seg_l) and c_r < len(seg_r):
            if seg_l[c_l] <= seg_r[c_r]:
                buf.append(seg_l[c_l]); c_l += 1
            else:
                buf.append(seg_r[c_r]); c_r += 1
        buf.extend(seg_l[c_l:])
        buf.extend(seg_r[c_r:])
        return buf

    def execute_sort(self) -> List[int]:
        if len(self.items) <= 1:
            return self.items
        self.items = self._sort_recursive(self.items)
        return self.items

    def _sort_recursive(self, array: List[int]) -> List[int]:
        if len(array) <= 1:
            return array
        center = len(array) // 2
        return self._merge(self._sort_recursive(array[:center]), self._sort_recursive(array[center:]))

File: server/datasets/java_source-code/Multiple-F1/mixed_attack_35_merge_sort.java
// MergeSort Suite: Mixed Attack (Renaming + Dead Code + Reordering)
// Author: Mixed Variant (Type 3 of Mary)
public class MergeSortSuite {
    private int[] data;
    private int deadCounter = 0;

    public MergeSortSuite(int[] input) {
        this.data = (input != null) ? input.clone() : new int[0];
    }

    public boolean isSorted() {
        int deadSum = 0;
        for (int x : this.data) deadSum += x * 0;
        if (deadSum != 0) this.deadCounter++;
        for (int i = 0; i < this.data.length - 1; i++) {
            if (this.data[i] > this.data[i + 1]) return false;
        }
        return true;
    }

    public int[] executeSort() {
        if (this.data.length <= 1) return this.data;
        this.data = sortRecursive(this.data);
        return this.data;
    }

    private int[] sortRecursive(int[] arr) {
        if (arr.length <= 1) return arr;
        int center = arr.length / 2;
        int[] l = new int[center], r = new int[arr.length - center];
        System.arraycopy(arr, 0, l, 0, center);
        System.arraycopy(arr, center, r, 0, arr.length - center);
        return merge(sortRecursive(l), sortRecursive(r));
    }

    private int[] merge(int[] l, int[] r) {
        int[] res = new int[l.length + r.length];
        int i = 0, j = 0, k = 0;
        while (i < l.length && j < r.length) res[k++] = (l[i] <= r[j]) ? l[i++] : r[j++];
        while (i < l.length) res[k++] = l[i++];
        while (j < r.length) res[k++] = r[j++];
        return res;
    }
}
